velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -225,6 +225,31 @@ TEST_F(DateTimeFunctionsTest, lastDay) {
   EXPECT_EQ(lastDayFunc(std::nullopt), std::nullopt);
 }
 
+TEST_F(DateTimeFunctionsTest, nextDay) {
+  const auto D = [](const std::string& date) -> std::optional<int32_t> {
+    return DATE()->toDays(date);
+  };
+
+  const auto nextDay = [&](const std::string& date,
+                           const std::optional<StringView> dow) {
+    std::vector<VectorPtr> flatVectors;
+    flatVectors.emplace_back(makeNullableFlatVector(
+        std::vector<std::optional<uint32_t>>{D(date)}, DATE()));
+    flatVectors.emplace_back(
+        makeNullableFlatVector(std::vector<std::optional<StringView>>{dow}));
```

Comment:
This is simple functon, you don't need to use VectorPtr to test, please take this as example, https://github.com/facebookincubator/velox/blob/main/velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp#L73

Reply:
simple interface does not support input columns such as Date and other types coexist

Reply:
No, it could. https://github.com/facebookincubator/velox/pull/4830/files#diff-f7042b580cdd92f6b4ba8b77183933cbb460c88a6deed81f767a35bb4d2cc40eR212

Reply:
Can we add a common test function in `FunctionBaseTest.h` to cover this case?

Reply:
> simple interface does not support input columns such as Date and other types coexist

@Ma-Jian1 Why do you say that? Can you open a GitHub issue if this is not working for you. It should work. CC: @laithsakka

Reply:
I can compile like this
```cpp
    return evaluateOnce<int32_t>(
        "next_day(c0, c1)", std::make_optional(date), dow);
```
but the signature is not right
when `date` type is `std::string`
```cpp
Reason: Scalar function signature is not supported: next_day(VARCHAR, VARCHAR). Supported signatures: (date,varchar) -> date.
```
or
when `date` type is `int32_t`
```cpp
Reason: Scalar function signature is not supported: next_day(INTEGER, VARCHAR). Supported signatures: (date,varchar) -> date.
```
Seems I should make `date` type `DATE`, which I can't construct.

Reply:
Try using a different version of evaluateOnce which allows to specify types of arguments:

```
template <typename ReturnType, typename Args>
  std::optional<ReturnType> evaluateOnce(
      const std::string& expr,
      const std::vector<std::optional<Args>>& args,
      const std::vector<TypePtr>& types,
      const std::optional<SelectivityVector>& rows = std::nullopt,
      const TypePtr& resultType = nullptr) {

return evaluateOnce<int32_t>( "next_day(c0, c1)", std::make_optional(date), dow, {DATE(), VARCHAR()});
```

Reply:
Sorry， it does not work.
```cpp
return evaluateOnce<int32_t>( "next_day(c0, c1)", {std::make_optional(date), dow}, {DATE(), VARCHAR()});
```
This should work if `date` and `dow` are the same type `Args`.

Reply:
What is the error you are getting?

Reply:
See https://github.com/facebookincubator/velox/pull/5926/ for a similar use case.